plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/StructuredTypeMemberChange.java

```diff
@@ -62,8 +59,8 @@ public Change perform(final IProgressMonitor pm) throws CoreException {
 
 		final StructuredType structuredTypeEditable = (StructuredType) affectedStruct.getTypeEntry().getTypeEditable();
 		for (final VarDeclaration varDeclaration : structuredTypeEditable.getMemberVariables()) {
-			final String typeName = varDeclaration.getTypeName();
-			if (typeName.equals(oldName)) {
+			if ((varDeclaration.getType() != null && typeEntry == varDeclaration.getType().getTypeEntry())
+					|| (typeEntry.getFullTypeName().equals(varDeclaration.getFullTypeName()))) {
```

Comment:
varDeclaration.getFullTypeName() won't give the full name if its package is imported on that location - only TypeEntry.getFullTypeName() is guaranteed to return the fully qualified name

Reply:
ok then I think it is better to remove the or part. Will do that.